src/driver/postgres/PostgresQueryRunner.ts

```diff
@@ -435,6 +435,17 @@ export class PostgresQueryRunner extends BaseQueryRunner implements QueryRunner
             }
         }
 
+        // if table have column with generated type, we must add the expression to the meta table
+        await Promise.all(table.columns
```

Comment:
Isn't it possible for the queries here to be inconsistently generated because of the `Promise.all`?  The order that they're generated in can change drastically because of it.

Reply:
Could you please elaborate more? As far as I know the order with Promise.all never changes. And even if it does, I think it isn't really an issue, because this line only affects the meta table. There it doesn't matter in which order they get inserted/removed.

Reply:
Ok, I think I known what you mean. This is an alternative approach:

```typescript
await Promise.all(table.columns
    .filter(column => column.generatedType === "STORED" && column.asExpression)
    .map(async column => {
        const tableName = await this.getTableNameWithSchema(table.name);
        const deleteQuery = new Query(`DELETE FROM typeorm_generation_meta WHERE table_name = $1 AND column_name = $2`, [tableName, column.name]);
        const insertQuery = new Query(`INSERT INTO typeorm_generation_meta(table_name, column_name, generation_expression) VALUES ($1, $2, $3)`, [tableName, column.name, column.asExpression]);
        return [ [deleteQuery, insertQuery], [deleteQuery] ];
    })
)
    .forEach(([ providedUpQueries, providedDownQueries ]) => {
        upQueries.push(...providedUpQueries);
        downQueries.push(...providedDownQueries);
    });
```

Reply:
I think what @imnotjames meant is to replace `Promise.all(table.columns` to `for (let column of table.columns)` in order to have a more predictable order of generated queries in migration.

Reply:
Ok, I can do this